src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -1397,15 +1397,18 @@ def aks_browse(cmd, client, resource_group_name, name, disable_browser=False,
                        'To use "az aks browse" first enable the add-on\n'
                        'by running "az aks enable-addons --addons kube-dashboard".')
 
-    proxy_url = 'http://{0}:{1}/'.format(listen_address, listen_port)
     _, browse_path = tempfile.mkstemp()
     # TODO: need to add an --admin option?
     aks_get_credentials(cmd, client, resource_group_name, name, admin=False, path=browse_path)
+
+    # Change to https if dashboard container is using https
+    protocol = 'http'
```

Comment:
can we move this down closer to where it is being used (but before the if statement that determines if it https - e.g. ln 1421)